src/averageAges.js

```diff
@@ -15,13 +15,21 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let mens = people.filter(person => person.sex === 'm');
+
+  if (century !== undefined) {
+    mens = mens.filter(person => Math.ceil(person.died / 100) === century);
+  }
+
+  const totalAge = mens.reduce((sum, person) => sum
+    + (person.died - person.born), 0);
+  const averageAge = parseFloat((totalAge / mens.length).toFixed(2));
+
+  return averageAge;
```

Comment:
The function name suggests it calculates the average age of men, but 'men' is a vague term. Consider renaming the parameter 'people' to a more descriptive name like 'records' or 'individuals'.
  Also, consider checking for 'undefined' instead of using a loose comparison with 'undefined'.